wandb/sdk/internal/incremental_table_util.py

```diff
@@ -1,23 +1,59 @@
+from __future__ import annotations
+
 import time
-from typing import TYPE_CHECKING
+from typing import TYPE_CHECKING, Any
 
 import wandb
 
 if TYPE_CHECKING:
     from wandb import Table
     from wandb.sdk.artifacts.artifact import Artifact
+
     from ..wandb_run import Run as LocalRun
 
 ART_TYPE = "wandb-run-incremental-table"
 
 
-def init_artifact(run: "LocalRun", sanitized_key: str) -> "Artifact":
+def handle_resumed_run(incr_table: Table, run: LocalRun, key: str):
+    """Handle different scenarios when a run is resumed.
+
+    Check the summary to see if there was an incremental table that was logged for
+    the key and assign the previous_increments_paths and increment_num to the table.
+    """
+    if not run.resumed or incr_table._resume_handled:
+        return
+
+    summary = run.summary
+
+    summary_from_key: dict[str, Any] | None = summary.get(key)
```

Comment:
This annotation is wrong, and you don't need it: at this point in the code, the type is `Any`. The `not isinstance(summary_from_key, dict)` check below makes typecheckers infer it's a `dict` if the `return` is bypassed.